### PR TITLE
Move JvmOptionParser to separate project

### PR DESCRIPTION
**Pull request by robbavey**
This commit moves the JvmOptionParser into its own gradle project.

This enables the JvmOptionParser to remain compatible with Java 1.8 to present a helpful error message to a user attempting to start Logstash using older versions of Java, while allowing the main Logstash code base to freely use idiomatic Java 11 features.

To test, attempt to run `bin/logstash`, `bin/logstash-plugin`, `bin/logstash-keystore`, etc on older Java versions, they should fail, giving a helpful error message, eg:

```
The minimum required Java version is 11; your Java version from [/Library/Java/JavaVirtualMachines/temurin-8.jdk/Contents/Home/jre] does not meet this requirement
```

### CODE REVIEW

**Comment by robbavey** [comment]
Waiting on #13670 to fix up license check

**Comment by robbavey** [comment]
CI: https://logstash-ci.elastic.co/job/elastic-logstash-main/185/
Windows CI: https://logstash-ci.elastic.co/job/elastic+logstash+main+multijob-windows-compatibility/147/
Unix compatibility CI: https://logstash-ci.elastic.co/job/elastic+logstash+main+multijob-unix-compatibility/138/
Main Acceptance: https://logstash-ci.elastic.co/job/elastic+logstash+main+multijob-acceptance/139/
Docker acceptance: https://logstash-ci.elastic.co/job/elastic+logstash+main+multijob-docker-acceptance/ (test cases are due to #14125)

File: build.gradle [hunk 1]
```diff
@@ -51,8 +52,8 @@ allprojects {
   apply plugin: 'idea'
   apply plugin: 'java-library'
 
-  project.sourceCompatibility = JavaVersion.VERSION_1_8
-  project.targetCompatibility = JavaVersion.VERSION_1_8
+  project.sourceCompatibility = JavaVersion.VERSION_11
+  project.targetCompatibility = JavaVersion.VERSION_11
```
Comment (yaauie) [inline build.gradle:56]: :tada: 😁

**Review by yaauie — APPROVED** [review]
LGTM. I have validated this locally and am VERY excited to finally be able to use modern(ish) Java features.

File: tools/jvm-options-parser/src/main/java/org/logstash/launchers/JavaVersion.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+/**
+ * Helper class to compare current version of JVM with a target version.
+ * Based on JavaVersion class from elasticsearch java version checker tool
+ */
+public class JavaVersion implements Comparable<JavaVersion> {
+
+    public static final JavaVersion CURRENT = parse(System.getProperty("java.specification.version"));
+    public static final JavaVersion JAVA_11 = parse("11");
+    private final List<Integer> version;
```
Comment (mashhurs) [inline tools/jvm-options-parser/src/main/java/org/logstash/launchers/JavaVersion.java:34]: minor: I would call this `versions` or `versionNumbers` (as it contains major, minot, etc.. numbers) as it is a collection.

File: tools/jvm-options-parser/src/main/java/org/logstash/launchers/JavaVersion.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            final int l = (i < left.size()) ? left.get(i) : 0;
+            final int r = (i < right.size()) ? right.get(i) : 0;
+            if (l < r) {
+                return -1;
+            }
+            if (r < l) {
+                return 1;
+            }
+        }
+        return 0;
```
Comment (mashhurs) [inline tools/jvm-options-parser/src/main/java/org/logstash/launchers/JavaVersion.java:77]: Left/Right approach also looks good. I think we can still improve readability using Javanized `compareTo` way?!
```
        Iterator<T> i1 = leftVersion.version.iterator();
        Iterator<T> i2 = rightVersion.version.iterator();
        int result;

        do {
            if (!i1.hasNext()) {
                return i2.hasNext() ? -1 : 0;
            }
            if (!i2.hasNext()) {
                return 1;
            }

            result = i1.next().compareTo(i2.next());
        } while (result == 0);

        return result;
```

File: tools/jvm-options-parser/src/main/java/org/logstash/launchers/JavaVersion.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        return new JavaVersion(version);
+    }
+
+    public static int majorVersion(final JavaVersion javaVersion) {
+        Objects.requireNonNull(javaVersion);
+        if (javaVersion.version.get(0) > 1) {
+            return javaVersion.version.get(0);
+        } else {
+            return javaVersion.version.get(1);
+        }
```
Comment (mashhurs) [inline tools/jvm-options-parser/src/main/java/org/logstash/launchers/JavaVersion.java:59]:
```
return javaVersion.version.get(0) > 1 javaVersion.version.get(0) :  javaVersion.version.get(1);
```

File: tools/jvm-options-parser/src/main/java/org/logstash/launchers/JavaVersion.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    static JavaVersion parse(final String value) {
+        if (value.matches("^0*[0-9]+(\\.[0-9]+)*$") == false) {
+            throw new IllegalArgumentException(value);
+        }
+
+        final List<Integer> version = new ArrayList<Integer>();
+        final String[] components = value.split("\\.");
+        for (final String component : components) {
+            version.add(Integer.valueOf(component));
+        }
```
Comment (mashhurs) [inline tools/jvm-options-parser/src/main/java/org/logstash/launchers/JavaVersion.java:49]:
```
final List<Integer> versionNumbers = Arrays.stream(value.split("\\."))
     .map(Integer::valueOf)
     .collect(Collectors.toList());
```

**Review by mashhurs — COMMENTED** [review]
My all comments are nits if we can improve in the future.

**Pull request merged**